src/main/java/org/generationcp/middleware/dao/GermplasmDAO.java

```diff
@@ -2067,4 +1903,83 @@ private void updateGroupSource(final List<Integer> gids, final Integer newGroupS
+				+ "		and cp.value in (select name from cvterm where cvterm_id in (" //
+				+ 			VariableType.GERMPLASM_PASSPORT.getId() + "," //
+				+ 			VariableType.GERMPLASM_ATTRIBUTE.getId() + ") ) " //
+				+ " and a.gid in (:" + GermplasmDAO.GIDS + ")" //
+				+ " group by v.cvterm_id");
+			sqlQuery.setParameter(GermplasmDAO.PROGRAM_UUID, programUUID);
+			sqlQuery.setParameterList(GermplasmDAO.GIDS, gids);
+			sqlQuery.setResultTransformer(Transformers.ALIAS_TO_ENTITY_MAP);
+			sqlQuery.addScalar(GermplasmDAO.VARIABLE_ID);
+			sqlQuery.addScalar(GermplasmDAO.VARIABLE_NAME);
+			sqlQuery.addScalar(GermplasmDAO.VARIABLE_DEFINITION);
+			sqlQuery.addScalar(GermplasmDAO.METHOD_ID);
+			sqlQuery.addScalar(GermplasmDAO.METHOD_NAME);
+			sqlQuery.addScalar(GermplasmDAO.METHOD_DEFINITION);
+			sqlQuery.addScalar(GermplasmDAO.PROPERTY_ID);
+			sqlQuery.addScalar(GermplasmDAO.PROPERTY_NAME);
+			sqlQuery.addScalar(GermplasmDAO.PROPERTY_DEFINITION);
+			sqlQuery.addScalar(GermplasmDAO.SCALE_ID);
+			sqlQuery.addScalar(GermplasmDAO.SCALE_NAME);
+			sqlQuery.addScalar(GermplasmDAO.SCALE_DEFINITION);
+			sqlQuery.addScalar(GermplasmDAO.VARIABLE_ALIAS);
+			sqlQuery.addScalar(GermplasmDAO.VARIABLE_EXPECTED_MIN);
+			sqlQuery.addScalar(GermplasmDAO.VARIABLE_EXPECTED_MAX);
+
+			final List<Map<String, Object>> queryResults = (List<Map<String, Object>>) sqlQuery.list();
+			final List<Variable> variables = new ArrayList<>();
+			for (final Map<String, Object> item : queryResults) {
+				final Variable variable =
+					new Variable(new Term(Util.typeSafeObjectToInteger(item.get(GermplasmDAO.VARIABLE_ID)), (String) item.get(GermplasmDAO.VARIABLE_NAME), (String) item.get(GermplasmDAO.VARIABLE_DEFINITION)));
+				variable.setMethod(new Method(new Term(Util.typeSafeObjectToInteger(item.get(GermplasmDAO.METHOD_ID)), (String) item.get(GermplasmDAO.METHOD_NAME), (String) item.get(GermplasmDAO.METHOD_DEFINITION))));
+				variable.setProperty(new Property(new Term(Util.typeSafeObjectToInteger(item.get(GermplasmDAO.PROPERTY_ID)), (String) item.get(GermplasmDAO.PROPERTY_NAME), (String) item.get(GermplasmDAO.PROPERTY_DEFINITION))));
+				variable.setScale(new Scale(new Term(Util.typeSafeObjectToInteger(item.get(GermplasmDAO.SCALE_ID)), (String) item.get(GermplasmDAO.SCALE_NAME), (String) item.get(GermplasmDAO.SCALE_DEFINITION))));
+
+				// Alias, Expected Min Value, Expected Max Value
+				final String pAlias = (String) item.get(GermplasmDAO.VARIABLE_ALIAS);
+				final String pExpMin = (String) item.get(GermplasmDAO.VARIABLE_EXPECTED_MIN);
+				final String pExpMax = (String) item.get(GermplasmDAO.VARIABLE_EXPECTED_MAX);
+
+				variable.setAlias(pAlias);
+				variable.setMinValue(pExpMin);
+				variable.setMaxValue(pExpMax);
+
+				variables.add(variable);
+			}
+			return variables;
+		} catch (final HibernateException e) {
+			final String message =
+				"Error with getGermplasmAttributeTypes(gids=" + gids + ") : " + e.getMessage();
```

Comment:
method name for log should be: `getGermplasmAttributeVariables`

Reply:
done.